### PR TITLE
Support vertical metrics tables

### PR DESCRIPTION
**Pull request by anthrotype**
Will eventually fix https://github.com/googlefonts/fontations/issues/103

### CODE REVIEW

File: read-fonts/src/tables/vmtx.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+//! The [vmtx (Vertical Metrics)](https://docs.microsoft.com/en-us/typography/opentype/spec/vmtx) table
+
+use font_types::Tag;
+
+/// 'vmtx'
+pub const TAG: Tag = Tag::new(b"vmtx");
```
Comment (cmyr) [inline read-fonts/src/tables/vmtx.rs:6]: okay, I have an idea for how we might want to structure this:

Here's what I think I would do:

- make the `hvmtx` and `hvhea` modules marked `pub(crate)` in `tables.rs`.
- manually reexport any shared structs (`LongMetric` in hvmtx, e.g.) from here, into each of the `hmtx` and `vmtx` modules
- add `pub` type aliases for `Hhea` and `Vhea` in the appropriate modules, with the appropriate doc strings

So in this file we would add something like,

```rust

pub use super::hvmtx::LongMetric;

/// The [vmtx (Vertical Metrics)](https://docs.microsoft.com/en-us/typography/opentype/spec/vmtx) table
pub type Vmtx = super::hvmtx::HVmtx;
```

and do this same trick in the others.

**Review by cmyr — APPROVED** [review]
Looks good! I have a minor suggestion for how we can structure the modules and exports, but overall I think this is the right approach.

File: resources/codegen_inputs/hvmtx.rs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    /// numberOfLongMetrics.
+    #[count($num_glyphs.saturating_sub($number_of_long_metrics) as usize)]
+    bearings: [i16],
+}
+
+record LongMetric {
+    /// Advance width/height, in font design units.
+    advance: u16,
+    /// Glyph leading (left/top) side bearing, in font design units.
+    sb: i16,
```
[the remark this answers is not in the document]
Comment (anthrotype) [reply resources/codegen_inputs/hvmtx.rs:21]: I believe I took the `sb` field name from harfbuzz' `LongMetric` struct:

https://github.com/harfbuzz/harfbuzz/blob/d1f445ec1e6518d2d135eca13e791a79a824025a/src/hb-ot-hmtx-table.hh#L59

but I can change it to `side_bearing` if you prefer so

File: read-fonts/src/tables/vmtx.rs (hunk 1 above)
Comment (anthrotype) [reply read-fonts/src/tables/vmtx.rs:6]: I can do that and add those type aliases, but what should methods like hhea() return, the shared `HVhea` struct or the type-aliased `Hhea` one?

```diff
- fn hhea(&self) -> Result<tables::hvhea::HVhea<'a>, ReadError>
+ fn hhea(&self) -> Result<tables::hhea::Hhea<'a>, ReadError>
```

Wouldn't we then basically hide from the API the fact that those tables are actually binary-encoded the same way and one can expect to use them interchangeably in user code?

File: resources/codegen_inputs/hvmtx.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cmyr) [reply resources/codegen_inputs/hvmtx.rs:21]: It's not a blocker, I just have a personal aversion to Very Short Names outside of specific common cases, so I defer to you.

File: read-fonts/src/tables/vmtx.rs (hunk 1 above)
Comment (cmyr) [reply read-fonts/src/tables/vmtx.rs:6]: that's a nice point. The final answer will depend on what we resolve for #109. let's leave this as is (no type aliases) for now, and we can revisit once we know our plan for the tag trait.

File: resources/codegen_inputs/hvmtx.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (anthrotype) [reply resources/codegen_inputs/hvmtx.rs]: renamed `side_bearing`

**Comment by anthrotype** [comment]
I just rebased this on main, I hope I didn't screw things up. At least the bots are happy 🤖

**Pull request merged**